code/modules/vehicles/mecha/mecha_construction_paths.dm

```diff
@@ -384,7 +384,9 @@
 	outer_plating_amount=1
 
 /datum/component/construction/mecha/gygax/action(datum/source, atom/used_atom, mob/user)
-	return INVOKE_ASYNC(src, PROC_REF(check_step), used_atom,user)
+	spawn(-1) //Please stop trying to get the return value from procs that can sleep
+		. = check_step(used_atom, user)
+	return .
```

Comment:
I do not like using this, straight up. spawn shouldn't be in "user space"

Reply:
We split the difference and make an `ASYNC` macro thats just `spawn(-1)`. Deal?

Reply:
the code should just be less stupid

Reply:
i agree but out of scope

Reply:
in scope spawn shouldn't be in user space

Reply:
its either spawn(), a couple UNLINT(), or PR close because im not refactoring mech code for this.